src/groups/semidirect_product_group.jl

```diff
@@ -88,296 +96,885 @@ See [HilgertNeeb:2012; Definition 9.2.22](@cite), first definition for more deta
+or the right group operation ``ρ_h(g) = g ∘ h`` with respect to its index ``g``.
+
+$(_doc_semidirect_sub_groups) Let ``τ`` denote a right group action. It here acts on the left.
+
+For the [`LeftSemidirectProductGroupOperation`](@ref) ``$(_math(:∘))`` on ``$(_tex(:Cal, "G")) ⋉ $(_tex(:Cal, "H"))`` we
+have
+
+```math
+    λ_{(g_1,h_1)}((g_2,h_2)) := (g_1,h_1) ∘ (g_2,h_2) = (g_1 ⋆ g_2, τ_{g_2^{-1}}(h_1) ⋄ h_2).
+```
+
+such that their differential reads for some ``(X, Y)`` from the Lie algebra that
+
+```math
+D λ_{(g_1,h_1)}(g_2,h_2)$(_tex(:bigl))[ (X,Y) $(_tex(:bigr))]
+  = $(_tex(:bigl))(
+  Dλ_{g_1}(g_2)[X],
+  Dλ_{τ_{g_2^{-1}}(h_1)}(h_2)[Y] + Dρ_{h_2}(τ_{g_2^{-1}}(h_1)) $(_tex(:bigl))[ D_{$(_tex(:Cal, "G"))}τ_{g_2^{-1}}(h_1)[X] $(_tex(:bigr))]
+  $(_tex(:bigr))).
+```
+
+For the [`RightSemidirectProductGroupOperation`](@ref) ``$(_math(:∘))`` on `` $(_tex(:Cal, "H")) ⋊ $(_tex(:Cal, "G"))`` we
+have
+
+```math
+    λ_{(h_1,g_1)}(h_2,g_2) := (h_1,g_1) ∘ (h_2,g_2) = (τ_{g_2^{-1}}(h_1) ⋄ h_2, g_1 ⋆ g_2).
+```
+
+such that their differential reads for some ``(Y, X)`` from the Lie algebra that
+
+```math
+D λ_{(h_1,g_1)}(h_2,g_2)$(_tex(:bigl))[ (Y, X) $(_tex(:bigr))]
+  = $(_tex(:bigl))(
+  Dλ_{τ_{g_2^{-1}}(h_1)}(h_2)[Y] + Dρ_{h_2}(τ_{g_2^{-1}}(h_1)) $(_tex(:bigl))[ D_{$(_tex(:Cal, "G"))}τ_{g_2^{-1}}(h_1)[X] $(_tex(:bigr))],
+  Dλ_{g_1}(g_2)[X],
+  $(_tex(:bigr))).
+```
+
+$(_semidirect_diff_compose_notation)
+"""
+diff_right_compose(
+    SDPG::LieGroup{𝔽, <:SemidirectProductGroupOperation{O1, O2, A, AO}, <:ProductManifold}, g, h, X
+) where {𝔽, O1, O2, A <: AbstractRightGroupActionType, AO <: ActionActsOnLeft}
+
+# An implementation for 1,3 (no inverse for left) and 5,7 (inverse for right)
+# in the computation of the element coming from the group action.
+function diff_right_compose!(
+        SDPG::LieGroup{𝔽, <:SemidirectProductGroupOperation{O1, O2, A, AO}, <:ProductManifold}, Y, g, h, X
+    ) where {𝔽, O1, O2, A <: AbstractGroupActionType, AO <: ActionActsOnLeft}
+    PM, G, H, a, g_ind, h_ind = _semidirect_parts(SDPG)
+    sdpg = LieAlgebra(SDPG)
+    # We use a naming close to the left variant: g = (gG, gH) = (g_1,h_1)
+    YG, YH = submanifold_component(sdpg, Y, Val(g_ind)), submanifold_component(sdpg, Y, Val(h_ind))
+    XG, XH = submanifold_component(sdpg, X, Val(g_ind)), submanifold_component(sdpg, X, Val(h_ind))
+    gG, gH = submanifold_component(SDPG, g, Val(g_ind)), submanifold_component(SDPG, g, Val(h_ind))
+    hG, hH = submanifold_component(SDPG, h, Val(g_ind)), submanifold_component(SDPG, h, Val(h_ind))
+    # For right actions we have to invert hG (g_2) - this allocates when it has to invert
+    hG_mod = _semidirect_maybe_inv(a, G, hG)
+    # one allocation for applying the action
+    # we need one allocation to compute the action (g_2 to h_1)
+    σg2mh1 = apply(a, hG_mod, gH)
+    # Step 1: Compute the argument for the second summand on H - use the memory of YH for the result
+    diff_group_apply!(A, YH, hG_mod, gH, XH)
```

Comment:
I think this should be:
```suggestion
    diff_group_apply!(a, YH, hG_mod, gH, XH)
```